samza-rest/src/main/java/org/apache/samza/monitor/SamzaMonitorService.java

```diff
@@ -43,16 +51,14 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(SamzaMonitorService.class);
   private static final SecureRandom RANDOM = new SecureRandom();
 
-  private final SchedulingProvider scheduler;
   private final SamzaRestConfig config;
   private final MetricsRegistry metricsRegistry;
-
+  private final List<ScheduledExecutorService> scheduledExecutors;
   public SamzaMonitorService(SamzaRestConfig config,
-                             MetricsRegistry metricsRegistry,
-                             SchedulingProvider schedulingProvider) {
+      MetricsRegistry metricsRegistry) {
     this.config = config;
     this.metricsRegistry = metricsRegistry;
-    this.scheduler = schedulingProvider;
+    scheduledExecutors = new ArrayList<>();
```

Comment:
Where is this list updated?

Reply:
this.scheduledExecutors for consistency.

Reply:
The list is updated in the createdScheduler, to stop all the scheduleders in the end when stop is called.